applications/GeoMechanicsApplication/custom_elements/interface_stress_state.cpp

```diff
@@ -0,0 +1,60 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#include "interface_stress_state.h"
+
+namespace Kratos
+{
+
+Matrix InterfaceStressState::CalculateBMatrix(const Matrix& rDN_DX, const Vector& rN, const Geometry<Node>& rGeometry) const
+{
+    if (rN.empty()) return {};
```

Comment:
We have to think about if we want to throw here, or return empty

Reply:
I vote for throw.
The B matrix of an empty thing is not defined, i.s.o. another empty thing.